deepset_cloud_sdk/service/files_service.py

```diff
@@ -213,21 +273,15 @@ async def upload(
         :recursive: If True, recursively upload all files in the folder.
         :raises TimeoutError: If blocking is True and the ingestion takes longer than timeout_s.
         """
-        logger.info("Getting valid files from file path. This may take up to a minute.")
-
-        all_files = self._get_file_paths(paths, recursive=recursive)
+        logger.info("Getting valid files from file path. This may take a few minutes.", recursive=recursive)
+        file_paths = []
 
-        file_paths = [
-            path
-            for path in all_files
-            if path.is_file() and ((path.suffix in [".txt", ".pdf"]) or path.name.endswith("meta.json"))
-        ]
-        if len(file_paths) < len(all_files):
-            logger.warning(
-                "Skipping files with unsupported file format.",
-                paths=paths,
-                skipped_files=len(all_files) - len(file_paths),
-            )
+        if show_progress:
+            with yaspin().arc as sp:
+                sp.text = "Finding uploadable files in the given paths"
+                file_paths = self._preprocess_paths(paths, spinner=None, recursive=recursive)
```

Comment:
I think we need to pass the spinner here, right ? Is it somehow possible to test that this is passed ?

Reply:
done